server/plugin.go

```diff
@@ -0,0 +1,139 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"net/http"
+	"strings"
+	"fmt"
+	"sync/atomic"
+
+	bbbAPI "github.com/blindsidenetworks/mattermost-plugin-bigbluebutton/server/bigbluebuttonapiwrapper/api"
+	"github.com/blindsidenetworks/mattermost-plugin-bigbluebutton/server/bigbluebuttonapiwrapper/dataStructs"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+	"github.com/robfig/cron"
+)
+
+type Plugin struct {
+
+	plugin.MattermostPlugin
+
+	c                            *cron.Cron
+	configuration                atomic.Value
+	Meetings                     []dataStructs.MeetingRoom
+	MeetingsWaitingforRecordings []dataStructs.MeetingRoom
+}
+
+//OnActivate runs as soon as plugin activates
+func (p *Plugin) OnActivate() error {
+	// we save all the meetings infos that are stored on in our database upon deactivation
+	// loads the details back so everything works
+	p.LoadMeetingsFromStore()
+
+	if err := p.OnConfigurationChange(); err != nil {
+		return err
+	}
+	config := p.config()
+	if err := config.IsValid(); err != nil {
+		return err
+	}
+
+	bbbAPI.SetAPI(config.BASE_URL+"/", config.SALT)
+
+	//every 2 minutes, look through active meetings and check if recordings are done
+	p.c = cron.New()
+	p.c.AddFunc("@every 2m", p.Loopthroughrecordings)
+	p.c.Start()
+
+	// register slash command '/bbb' to create a meeting
+	return p.API.RegisterCommand(&model.Command{
+		Trigger:          "bbb",
+		AutoComplete:     true,
+		AutoCompleteDesc: "Create a BigBlueButton meeting",
+	})
+}
+
+
+//following method is to create a meeting from '/bbb' slash command
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+
+	meetingpointer := new(dataStructs.MeetingRoom)
+	err := p.PopulateMeeting(meetingpointer, nil, "") 
+	if err != nil {
+		return nil, model.NewAppError("ExecuteCommand", "Please provide a 'Site URL' in Settings > General > Configuration", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	p.createStartMeetingPost(args.UserId, args.ChannelId, meetingpointer)
+	p.Meetings = append(p.Meetings, *meetingpointer)
+	return &model.CommandResponse{}, nil
+
+}
+
+//this is the router to handle our server calls
+//methods are all in responsehandlers.go
+func (p *Plugin) ServeHTTP(c *plugin.Context,w http.ResponseWriter, r *http.Request) {
+
+	config := p.config()
+	if err := config.IsValid(); err != nil {
+		http.Error(w, "This plugin is not configured.", http.StatusNotImplemented)
+		return
+	}
+
+	path := r.URL.Path
+	if path == "/joinmeeting" {
```

Comment:
Maybe use https://github.com/gorilla/mux?

Reply:
Any specific benefits of switching to mux here?

Reply:
Cleaner code mostly